src/main/java/com/spring/demo_relationship/service/user/UserServiceImpl.java

```diff
@@ -1,31 +1,26 @@
 package com.spring.demo_relationship.service.user;
 
-import com.spring.demo_relationship.dto.UserDto;
 import com.spring.demo_relationship.exceptions.ResourceNotFoundException;
 import com.spring.demo_relationship.models.DoctorProfile;
+import com.spring.demo_relationship.models.Role;
 import com.spring.demo_relationship.models.UserEntity;
 import com.spring.demo_relationship.repository.DoctorProfileRepository;
 import com.spring.demo_relationship.repository.UserRepository;
 import lombok.RequiredArgsConstructor;
-import org.springframework.data.crossstore.ChangeSetPersister;
-import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
-import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
     private final UserRepository userRepository;
     private final DoctorProfileRepository doctorProfileRepository;
 
-    private final UserEntity existingUser = getCurrentUser();
 
-    @Override
-    public UserEntity getProfile(String email) {
+    private UserEntity getProfile(String email) {
         UserEntity user = userRepository.findByEmail(email);
         if (user == null) {
             throw new ResourceNotFoundException("User with email : " + email + " not found");
```

Comment:
try here to create your own exception see this link : 
https://github.com/anasabbal/mini-uber-microservice/blob/master/core-comm/src/main/java/com/nas/core/exception/ExceptionPayloadFactory.java